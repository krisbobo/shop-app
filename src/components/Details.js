import React from 'react';
import style from './Details.module.css';

const Details = () => (
  <div className={style.app}>
    <div className={style.container}>
      <div className={style.top}>
        <div className={style.location}>
          <p>Houston</p>
        </div>
        <div className="temp">
          <h1>79°F</h1>
        </div>
        <div className={style.description}>
          <p>Clouds</p>
        </div>
      </div>
      <div className={style.bottom}>
        <div className="humidity">
          <p className="bold">50%</p>
          <p>Humidity</p>
        </div>
        <div className="feels">
          <p className="bold">79°F</p>
          <p>Feels Like</p>
        </div>
        <div className="wind">
          <p className="bold">5MPH</p>
          <p>Wind Speed</p>
        </div>
      </div>
    </div>
  </div>

);

export default Details;
