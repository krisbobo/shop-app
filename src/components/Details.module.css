@import url('https://fonts.googleapis.com/css?family=lato:300,400,500,700&display=swap');

* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
}

p {
  font-size: 1.3rem;
}

h1 {
  font-size: 6rem;
}

.app {
  width: 100%;
  height: 100%;
  position: relative;
  background-color: rgb(255, 71, 135, 0.5);
  color: #fff;
}

.app:before {
  content: '';
  background: url('../assets/photo4.jpg') no-repeat center center/cover;
  position: absolute;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  z-index: -1;
}

.container {
  max-width: 700px;
  height: 700px;
  margin: auto;
  padding: 0 1rem;
  position: relative;
  top: 10%;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}

.app .top {
  width: 100%;
  margin: 1rem auto;
  padding: 2rem 1rem;
}

.app .description {
  position: relative;
  right: -90%;
  transform-origin: 0 0;
  transform: rotate(270deg);
}

.app .bottom {
  display: flex;
  justify-content: space-between;
  text-align: center;
  width: 100%;
  margin: 1rem auto;
  padding: 1.5rem;
  border-radius: 12px;
  background-color: rgba(255, 255, 255, 0.2);
}

.bold {
  font-weight: 700;
}