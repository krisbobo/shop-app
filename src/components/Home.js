import React from 'react';
// import { useDispatch, useSelector } from 'react-redux';
// import { getCities } from '../redux/home/home';
import style from './Home.module.css';

const Home = () => (
  <div className={style.app}>
    <nav>
      <span>My weather</span>
      <span>Home</span>
    </nav>
    <div className={style.container}>
      <div className={style.search}>
        <input placeholder="Enter your city" />
      </div>
      <div className={style.cities}>
        <div className={style.city}>
          <div className={style.cityName}>
            <p>City1</p>
            <p>Country1</p>
          </div>
        </div>
        <div className={style.city}>
          <div className={style.cityName}>
            <p>City2</p>
            <p>Country2</p>
          </div>
        </div>
        <div className={style.city}>
          <div className={style.cityName}>
            <p>City3</p>
            <p>Country3</p>
          </div>
        </div>
        <div className={style.city}>
          <div className={style.cityName}>
            <p>City4</p>
            <p>Country4</p>
          </div>
        </div>

      </div>
    </div>
  </div>
);

export default Home;

// const dispatch = useDispatch();
// const cities = useSelector((state) => state.home.cities);
// useEffect(() => {
//   dispatch(getCities(cities));
// }, [dispatch, cities]);
