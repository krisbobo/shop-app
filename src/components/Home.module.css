* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  overflow-x: hidden;
  max-width: 100%;
}

p {
  font-size: 1.2rem;
}

span {
  font-size: 1.4rem;
}

nav {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 1.5rem;
}

.app {
  display: flex;
  flex-direction: column;
  width: 100%;
  height: 100%;
  position: relative;
  background-color: rgb(255, 71, 135, 0.5);
  color: rgb(251, 253, 254);
}

.app:before {
  content: '';
  position: absolute;
  background: url('../assets/photo4.jpg') no-repeat center center/cover;
  width: 100%;
  height: 100%;
  top: 0;
  left: 0;
  z-index: -1;
}

.container {
  display: flex;
  flex-direction: column;
  margin-top: 2rem;
  gap: 2rem;
}

.app .search {
  text-align: center;
  padding: 1rem;
}

.app input {
  font-size: 1.2rem;
  padding: 0.7rem 1.4rem;
  border-radius: 25px;
  background: rgba(255, 255, 255, 0.2);
  color: #f8f8f8;
  border: none;
}

::placeholder {
  color: #f8f8f8;
}

.app .cities {
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  height: 100%;
  background-color: rgb(225, 61, 117, 0.6);
}

.app .city {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  padding: 1rem;
}

.app .city:nth-child(4n - 2) {
  background-color: rgb(209, 58, 108);
}

.app .city:nth-child(4n - 1) {
  background-color: rgb(209, 58, 108);
}

.city .cityName {
  font-size: 1.4rem;
  font-weight: 700;
  text-align: right;
  padding: 0.3rem 1rem;
}
