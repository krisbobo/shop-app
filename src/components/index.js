export { default as Navbar } from './navbar';
export { default as Products } from './products';
export { default as Product } from './product';
