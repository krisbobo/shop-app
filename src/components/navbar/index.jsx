import { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import {
  NavLink, Outlet, useNavigate, useLocation,
} from 'react-router-dom';
import { GiHamburgerMenu } from 'react-icons/gi';
import { MdClose } from 'react-icons/md';
import {
  getCategories,
  isActive,
  getProductsByCategory,
  getProducts,
  swapCategoryTitle,
} from '../../redux/homeSlice';
import { isActive as isOpened } from '../../redux/detailSlice';

import {
  Logo,
  Container,
  Header,
  NavLinks,
  Li,
  BurgerButton,
  BurgerInnerContainer,
  CloseButton,
  BurgerNavContainer,
  BackButton,
} from './styles/navbar';

const Navbar = () => {
  const dispatch = useDispatch();
  const navigate = useNavigate();
  const location = useLocation();
  const homeReducer = useSelector((state) => state.homeReducer);
  const { categories, isActive: isOpen, categoryTitle } = homeReducer;
  useEffect(() => {
    if (categories.length === 0) dispatch(getCategories());
  }, [categories.length, dispatch]);

  const handleBurgerButtonClick = () => {
    dispatch(isActive());
    dispatch(isOpened());
  };

  const handleNavLinksClick = (event, category) => {
    dispatch(getProductsByCategory(category));

    const categoryTitle = category.charAt(0).toUpperCase() + category.slice(1);

    if (!location.pathname.includes('/details')) {
      dispatch(swapCategoryTitle(categoryTitle));
      dispatch(isActive());
    }

    if (location.pathname.includes('/details')) {
      dispatch(swapCategoryTitle(categoryTitle));
      dispatch(isActive());
    }

    navigate('/');
  };

  const category = categories?.map((category) => {
    const isSelected = category === categoryTitle.charAt(0).toLowerCase() + categoryTitle.slice(1);
    const isDetails = location.pathname.includes('/details');

    return (
      <Li
        key={category}
        onClick={(event) => handleNavLinksClick(event, category)}
        isDetails={isDetails}
        isSelected={isSelected}
        isOpen={isOpen}
      >
        {category}
      </Li>
    );
  });

  const getAllProducts = () => {
    dispatch(getProducts());
    dispatch(swapCategoryTitle('All Products'));
  };

  return (
    <>
      <Header isActive={isOpen}>
        <Container>
          <BurgerInnerContainer isVisible={location.pathname.includes('/details')}>
            {location.pathname.includes('/details') ? (
              <BackButton onClick={() => navigate('/')}>Back</BackButton>
            ) : (
              ''
            )}

            <Logo onClick={getAllProducts}>
              <NavLink to="/">Shop</NavLink>
            </Logo>
            <BurgerButton onClick={handleBurgerButtonClick}>
              <GiHamburgerMenu />
            </BurgerButton>
          </BurgerInnerContainer>
          <BurgerNavContainer isActive={isOpen}>
            <CloseButton onClick={handleBurgerButtonClick}>
              <MdClose />
            </CloseButton>
            <NavLinks>{category}</NavLinks>
          </BurgerNavContainer>
        </Container>
      </Header>

      <Outlet />
    </>
  );
};

export default Navbar;
