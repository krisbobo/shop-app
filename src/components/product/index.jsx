import PropTypes from 'prop-types';
import { useNavigate } from 'react-router-dom';

import {
  Container,
  Image,
  Title,
  Price,
  Rating,
  DetailsContainer,
  ImageContainer,
  InnerDetailsContainer,
} from './styles/product';

const Product = ({ product }) => {
  const {
    title, image, price, rating,
  } = product;
  const navigate = useNavigate();

  const handleClick = () => {
    navigate(`/details/product/${product.id}`);
  };

  return (
    <Container onClick={handleClick}>
      <ImageContainer>
        <Image src={image} alt={title} />
      </ImageContainer>
      <DetailsContainer>
        <Title>{title}</Title>
        <InnerDetailsContainer>
          <Price>
            $
            {price}
          </Price>
          <Rating>
            Rate
            {rating.rate}
          </Rating>
        </InnerDetailsContainer>
      </DetailsContainer>
    </Container>
  );
};

Product.propTypes = {
  product: PropTypes.shape({
    id: PropTypes.number,
    title: PropTypes.string,
    image: PropTypes.string,
    price: PropTypes.number,
    rating: PropTypes.shape({
      rate: PropTypes.number,
      count: PropTypes.number,
    }),
  }).isRequired,
};

export default Product;
