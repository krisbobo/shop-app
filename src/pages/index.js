export { default as Home } from './home/index';
export { default as Details } from './details/index';
