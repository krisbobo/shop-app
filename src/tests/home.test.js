import { render } from '@testing-library/react';
import { useSelector } from 'react-redux';
import { Products, Product } from '../components';
import { Home } from '../pages';

jest.mock('react-redux');
jest.mock('react-router-dom');

describe('Test Home components rendering', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('should render Products component', () => {
    useSelector.mockReturnValue({
      isLoading: false,
      products: [
        {
          category: 'electronics',
          description: 'test',
          id: 14,
          image: 'https://fakestoreapi.com/img/81Zt42ioCgL._AC_SX679_.jpg',
          price: 999.99,
          title: 'title',
          rating: {
            rate: 2.2,
            count: 140,
          },
        },
      ],
      isActive: false,
    });
    const tree = render(<Products />);
    expect(tree).toMatchSnapshot();
  });

  test('should render Home page', () => {
    useSelector.mockReturnValue({
      isActive: false,
      categoryTitle: 'electronics',
      products: [
        {
          category: 'electronics',
          description: 'test',
          id: 14,
          image: 'https://fakestoreapi.com/img/81Zt42ioCgL._AC_SX679_.jpg',
          price: 999.99,
          title: 'title',
          rating: {
            rate: 2.2,
            count: 140,
          },
        },
      ],
    });

    const tree = render(<Home />);
    expect(tree).toMatchSnapshot();
  });

  test('should render Product', () => {
    const product = {
      category: 'electronics',
      description: 'test',
      id: 14,
      image: 'https://fakestoreapi.com/img/81Zt42ioCgL._AC_SX679_.jpg',
      price: 999.99,
      title: 'title',
      rating: {
        rate: 2.2,
        count: 140,
      },
    };
    const tree = render(<Product product={product} />);
    expect(tree).toMatchSnapshot();
  });
});
