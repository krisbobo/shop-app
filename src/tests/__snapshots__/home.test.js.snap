// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Home components rendering should render Home page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="home__Main-sc-1osvp4n-0 kZiaqO"
      >
        <h2
          class="home__Title-sc-1osvp4n-1 lceBYY"
        >
          electronics (1)
        </h2>
        <section
          class="products__Container-sc-1h2mq1j-0 jyLuIM"
        >
          <ul
            class="products__List-sc-1h2mq1j-1 dvYPYr"
          >
            <li
              class="product__Container-sc-1e87vg-0 kyufYN"
            >
              <div
                class="product__ImageContainer-sc-1e87vg-3 AWPSs"
              >
                <img
                  alt="title"
                  class="product__Image-sc-1e87vg-4 ciJYtL"
                  src="https://fakestoreapi.com/img/81Zt42ioCgL._AC_SX679_.jpg"
                />
              </div>
              <div
                class="product__DetailsContainer-sc-1e87vg-1 cZkspc"
              >
                <h2
                  class="product__Title-sc-1e87vg-5 dPtzuD"
                >
                  title
                </h2>
                <div
                  class="product__InnerDetailsContainer-sc-1e87vg-2 irHFSv"
                >
                  <span
                    class="product__Price-sc-1e87vg-6 itKTaJ"
                  >
                    $
                    999.99
                  </span>
                  <span
                    class="product__Rating-sc-1e87vg-7 gLITru"
                  >
                    Rate
                    2.2
                  </span>
                </div>
              </div>
            </li>
          </ul>
        </section>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="home__Main-sc-1osvp4n-0 kZiaqO"
    >
      <h2
        class="home__Title-sc-1osvp4n-1 lceBYY"
      >
        electronics (1)
      </h2>
      <section
        class="products__Container-sc-1h2mq1j-0 jyLuIM"
      >
        <ul
          class="products__List-sc-1h2mq1j-1 dvYPYr"
        >
          <li
            class="product__Container-sc-1e87vg-0 kyufYN"
          >
            <div
              class="product__ImageContainer-sc-1e87vg-3 AWPSs"
            >
              <img
                alt="title"
                class="product__Image-sc-1e87vg-4 ciJYtL"
                src="https://fakestoreapi.com/img/81Zt42ioCgL._AC_SX679_.jpg"
              />
            </div>
            <div
              class="product__DetailsContainer-sc-1e87vg-1 cZkspc"
            >
              <h2
                class="product__Title-sc-1e87vg-5 dPtzuD"
              >
                title
              </h2>
              <div
                class="product__InnerDetailsContainer-sc-1e87vg-2 irHFSv"
              >
                <span
                  class="product__Price-sc-1e87vg-6 itKTaJ"
                >
                  $
                  999.99
                </span>
                <span
                  class="product__Rating-sc-1e87vg-7 gLITru"
                >
                  Rate
                  2.2
                </span>
              </div>
            </div>
          </li>
        </ul>
      </section>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test Home components rendering should render Product 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="product__Container-sc-1e87vg-0 kyufYN"
      >
        <div
          class="product__ImageContainer-sc-1e87vg-3 AWPSs"
        >
          <img
            alt="title"
            class="product__Image-sc-1e87vg-4 ciJYtL"
            src="https://fakestoreapi.com/img/81Zt42ioCgL._AC_SX679_.jpg"
          />
        </div>
        <div
          class="product__DetailsContainer-sc-1e87vg-1 cZkspc"
        >
          <h2
            class="product__Title-sc-1e87vg-5 dPtzuD"
          >
            title
          </h2>
          <div
            class="product__InnerDetailsContainer-sc-1e87vg-2 irHFSv"
          >
            <span
              class="product__Price-sc-1e87vg-6 itKTaJ"
            >
              $
              999.99
            </span>
            <span
              class="product__Rating-sc-1e87vg-7 gLITru"
            >
              Rate
              2.2
            </span>
          </div>
        </div>
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="product__Container-sc-1e87vg-0 kyufYN"
    >
      <div
        class="product__ImageContainer-sc-1e87vg-3 AWPSs"
      >
        <img
          alt="title"
          class="product__Image-sc-1e87vg-4 ciJYtL"
          src="https://fakestoreapi.com/img/81Zt42ioCgL._AC_SX679_.jpg"
        />
      </div>
      <div
        class="product__DetailsContainer-sc-1e87vg-1 cZkspc"
      >
        <h2
          class="product__Title-sc-1e87vg-5 dPtzuD"
        >
          title
        </h2>
        <div
          class="product__InnerDetailsContainer-sc-1e87vg-2 irHFSv"
        >
          <span
            class="product__Price-sc-1e87vg-6 itKTaJ"
          >
            $
            999.99
          </span>
          <span
            class="product__Rating-sc-1e87vg-7 gLITru"
          >
            Rate
            2.2
          </span>
        </div>
      </div>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test Home components rendering should render Products component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="products__Container-sc-1h2mq1j-0 jyLuIM"
      >
        <ul
          class="products__List-sc-1h2mq1j-1 dvYPYr"
        >
          <li
            class="product__Container-sc-1e87vg-0 kyufYN"
          >
            <div
              class="product__ImageContainer-sc-1e87vg-3 AWPSs"
            >
              <img
                alt="title"
                class="product__Image-sc-1e87vg-4 ciJYtL"
                src="https://fakestoreapi.com/img/81Zt42ioCgL._AC_SX679_.jpg"
              />
            </div>
            <div
              class="product__DetailsContainer-sc-1e87vg-1 cZkspc"
            >
              <h2
                class="product__Title-sc-1e87vg-5 dPtzuD"
              >
                title
              </h2>
              <div
                class="product__InnerDetailsContainer-sc-1e87vg-2 irHFSv"
              >
                <span
                  class="product__Price-sc-1e87vg-6 itKTaJ"
                >
                  $
                  999.99
                </span>
                <span
                  class="product__Rating-sc-1e87vg-7 gLITru"
                >
                  Rate
                  2.2
                </span>
              </div>
            </div>
          </li>
        </ul>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="products__Container-sc-1h2mq1j-0 jyLuIM"
    >
      <ul
        class="products__List-sc-1h2mq1j-1 dvYPYr"
      >
        <li
          class="product__Container-sc-1e87vg-0 kyufYN"
        >
          <div
            class="product__ImageContainer-sc-1e87vg-3 AWPSs"
          >
            <img
              alt="title"
              class="product__Image-sc-1e87vg-4 ciJYtL"
              src="https://fakestoreapi.com/img/81Zt42ioCgL._AC_SX679_.jpg"
            />
          </div>
          <div
            class="product__DetailsContainer-sc-1e87vg-1 cZkspc"
          >
            <h2
              class="product__Title-sc-1e87vg-5 dPtzuD"
            >
              title
            </h2>
            <div
              class="product__InnerDetailsContainer-sc-1e87vg-2 irHFSv"
            >
              <span
                class="product__Price-sc-1e87vg-6 itKTaJ"
              >
                $
                999.99
              </span>
              <span
                class="product__Rating-sc-1e87vg-7 gLITru"
              >
                Rate
                2.2
              </span>
            </div>
          </div>
        </li>
      </ul>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
